kernel/src/core_module.h

```diff
@@ -3,19 +3,30 @@
 
 #include <linux/proc_fs.h>
 
-typedef struct plugin_ops_s {
+typedef enum plugin_state_e {
+    PLUGIN_STATE_INITIALIZED,
+    PLUGIN_STATE_RUNNING,
+    PLUGIN_STATE_STOPPED,
+    PLUGIN_STATE_DESTROYED,
+    PLUGIN_STATE_ERROR,
+} plugin_state_t;
```

Comment:
name implies context
```suggestion
typedef enum plugin_state_e {
    INITIALIZED,
    RUNNING,
    STOPPED,
    DESTROYED,
    ERROR,
} plugin_state_t;
```